registration/include/pcl/registration/impl/sample_consensus_prerejective.hpp

```diff
@@ -305,20 +318,28 @@ SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::computeTransfor
 template <typename PointSource, typename PointTarget, typename FeatureT>
 void
 SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::getFitness(
-    pcl::Indices& inliers, float& fitness_score)
+    pcl::Indices& inliers, float& fitness_score) const
+{
+  fitness_score = getFitness(final_transformation_, inliers);
+}
+
+template <typename PointSource, typename PointTarget, typename FeatureT>
+float
+SampleConsensusPrerejective<PointSource, PointTarget, FeatureT>::getFitness(
+    const Eigen::Matrix4f& transformation, pcl::Indices& inliers) const
 {
   // Initialize variables
   inliers.clear();
   inliers.reserve(input_->size());
-  fitness_score = 0.0f;
+  float fitness_score = 0.0f;
 
   // Use squared distance for comparison with NN search results
   const float max_range = corr_dist_threshold_ * corr_dist_threshold_;
 
-  // Transform the input dataset using the final transformation
+  // Transform the input dataset using the transformation
   PointCloudSource input_transformed;
   input_transformed.resize(input_->size());
-  transformPointCloud(*input_, input_transformed, final_transformation_);
+  transformPointCloud(*input_, input_transformed, transformation);
 
   // For each point in the source dataset
```

Comment:
nn_indices and nn_dists could be moved outside the for loop to reduce (de)allocations.